### DISCUSSION TITLE
How to integrate recorded map?

### DISCUSSION THREAD

**Discussion by michaelbr51** [osmandapp/OsmAnd#16170] [Maps Related]
The place where I live currently map is incomplete (lack of roads). I recorded some of the roads and I've asked to show on my map, the problem I'm having is everytime I'm travelling the recorded road, the app keep saying "... recalculating the route...", here are my questions:
1) how can I integrate those missing roads into normal map? or disable the warning when travelling on recorded track?
2) how can I suggest improvements to route traveling? For instance, during my trip, a highway split into 2, but the app does not tell me which one to take, etc.

Thanks for your suggestion/comment
Michael

**Comment by vshcherb** [comment]
You can upload GPX tracks to OSM https://wiki.openstreetmap.org/wiki/Upload_GPS_tracks and then use for editing. It looks like there are issues with map data but you need to get acquainted with OSM Editing.
